chi/network.py

```diff
@@ -412,28 +414,35 @@ def get_router_id(name) -> str:
     return _resolve_id('routers', name)
 
 
-def create_router(router_name, gateway=False) -> dict:
-    """Create a router, with or without a public gateway.
-
-    Args:
-        router_name (str): The new router name.
-        gateway (bool): Whether to add a gateway to the public Internet on
-            the router. If no public gateway is requested, subnets connected
-            to this router will not have NAT to the Internet. (Default False).
-
-    Returns:
-        The created router representation.
-    """
-    router = {
-        'name': router_name,
-        'admin_state_up': True,
-    }
-
-    if gateway:
-        public_net_id = get_network_id(PUBLIC_NETWORK)
-        router['external_gateway_info'] = {'network_id': public_net_id}
-
-    return neutron().create_router(body=router)
+def create_router(router_name, gw_network_name=None) -> dict:
+    ''' 
+    Create a router with or without a public gateway. 
+    
+    Parameters
+    ----------
+    router_name : str
+        Name of the new router.
+    gw_network_name: str
+        Name of the external gateway network (i.e. the network that connects to the Internet). 
+        Chameleon gateway network is 'public'. Default: None
+    '''
+    request = {}
+    if gw_network_name:
+        public_net_id= get_network_id(gw_network_name)
+        
+        #Create Router
+        request = {'router': {'name': router_name,
+                              'admin_state_up': True,
+                              'external_gateway_info': {"network_id": public_net_id},
+                             }}
+    else:
+        #Create Router without gateway
+        request = {'router': {'name': router_name,
+                              'admin_state_up': True,
+                             }}
+        
+    router = neutron().create_router(request)
+    return router
```

Comment:
I think this is a bit cleaner to express if the `router` payload is built up with the optional pieces rather than re-declaring the keys that are sent in either case. Plus, it reduces the likelihood that we update the `name` in one location but not the other, for example.

```suggestion
    router = {"name": router_name, "admin_state_up": True}
    
    if gw_network_name:
        router["external_gateway_info"] = {"network_id": get_network_id(gw_network_name)}
        
    response = neutron().create_router(body={"router": router})
    return response["router"]
```

It looks like this endpoint returns a nested structure: https://docs.openstack.org/api-ref/network/v2/index.html?expanded=create-router-detail#id207, so we need to unwrap it, probably?

Reply:
Yeah, that makes sense - I wasn't sure if the router structure was being embedded because of other things that could be contained in the wrapper. I've went ahead and pasted in your changes and will be testing related methods.

Reply:
Everything checks out. This has been pushed.